specs/agents/tracing-instrumentation-aws-lambda.md

```diff
@@ -132,6 +132,68 @@ In version 2.0, the `${event.requestContext.routeKey}` can have the format `GET
 
 If `use_path_as_transaction_name` is applicable and set to `true`, use `${event.requestContext.http.method} ${event.requestContext.http.path}` as the transaction name.
 
+### Elastic Load Balancer (ELB)
+
+Elastic Load Balancer (ELB) can be attached directly to lambda, without the use
+of API Gateway. In this case the `event` object will be structured differently.
+
+The agent should use the information in the request and response objects to
+fill the HTTP context (`context.request` and `context.response`) fields in the
+same way it is done for HTTP transactions.
+[Request/Response Docs](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/lambda-functions.html#receive-event-from-load-balancer)
+
+In particular, agents must use the `event.headers` to retrieve the
+`traceparent` and the `tracestate` and use them to start the transaction for
+the lambda function execution.
+
+In addition the following fields should be set for ELB-based Lambda functions:
+
+Field | Value | Description | Source
+---   | ---   | ---         | ---
+`type` | `request`| Transaction type: constant value for ELB. | -
+`name` | e.g. `GET /prod/proxy/{proxy+}` | Transaction name: Http method followed by a whitespace and the (resource) path. See section below. | -
+`transaction.result` | `HTTP Xxx` / `success` | `HTTP 5xx` if there was a function error. If the [invocation response has a "statusCode" field](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/lambda-functions.html#respond-to-load-balancer), then set to `HTTP Xxx` based on the status code, otherwise `success`. | Error or `response.statusCode`.
+`faas.trigger.type` | `http` | Constant value for ELB. | -
+`faas.trigger.request_id` | e.g. `Root=1-5bdb40ca-556d8b0c50dc66f0511bf520` | AWS Trace ID of the ELB request. | `event.headers.x-amzn-trace-id`
```

Comment:
How about leaving the `faas.trigger.request_id` empty in case of elb? Otherwise, we'd need to have this as a fallback for other trigger types as well.

For consistency, let's keep it empty here. WDYT?

Reply:
Seems fine to me.